File: client/src/services/dtos/shared-post-dto.ts
import PostDto from './post-dto';
import { UserDto } from './user-dto';

export interface SharedPostDto
  extends Pick<
    PostDto,
    | 'id'
    | 'anonymous'
    | 'section'
    | 'uploadTime'
    | 'mediaType'
    | 'mediaUrl'
    | 'contentType'
    | 'title'
    | 'text'
  > {
  readonly originalPoster: UserDto;
}

File: back-end/src/main/java/com/huyphan/dtos/SharedPostDto.java
package com.huyphan.dtos;

import com.huyphan.models.enums.PostContentType;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class SharedPostDto {

    private Long id;

    private UserDto originalPoster;

    private boolean anonymous;

    private SectionDto section;

    private String uploadTime;

    private PostContentType contentType;

    private String text;

    private String title;

    private String mediaUrl;

    private String mediaType;
}
